docs_src/microservices/device/V2Migration.md

```diff
@@ -0,0 +1,85 @@
+# V2 Migration Guide
+
+!!! edgey "EdgeX 2.0"
+For the EdgeX 2.0 (Ireland) release there are many backward breaking changes. These changes require custom Device Services and custom device profiles to be migrated. This section outlines the necessary steps for this migration.
+
+## Custom Device Services
+
+### Configuration
+
+The migration of any Device Service's configuration starts with migrating configuration common to all EdgeX services. See the [V2 Migration of Common Configuration](../../configuration/V2MigrationCommonConfig) section for details. The remainder of this section focuses on configuration specific to Device Services.
+
+#### Device
+1. Remove `ImitCmd`, `ImitCmdArgs`, `RemoveCmd` and `RemoveCmdArgs`
+2. Add `UseMessageBus` to determine events should be published to MessageBus or sent by REST call.
+3. Add `DevicesDir` and `ProfilesDir` as an indication of where to load the device profiles and pre-defined devices. Convention is to put them under `/res` folder:
+```
+(where binary resides ...)
++- res
+|  +- devices
+|    +- device1.toml
+|    +- device2.toml
+|  +- profiles
+|    +- profile1.yml
+|    +- profile2.yml
+|  +- configuration.toml
+|  +- ...
++- main.go
+
+```
+
+#### MessageQueue
+See the [Device Service MessageQueue](../../design/adr/013-Device-Service-Events-Message-Bus.md#device-services) section for details
+
+### Code
+
+#### Dependencies
+You first need to update all the dependencies to v2 major release: update `go.mod`, all the import path and data models.
+
+#### CommandValue
+You might then notice the original API to create CommandValue is no longer working.
+In V2 we refactor all those API functions to create CommandValue of different type to a simple generic function:
+
+!!! example "Create CommandValue with string Type"
+    ```go
+    cv, err := models.NewCommandValue(deviceResourceName, v2.ValueTypeString, "foobar")
+    if err != nil {
+        ...
+    }
+    cv.Origin = time.Now().Unixnano()
+    ```
+
+The 3rd argument in the function must be able to cast into the Type defined in 2nd argument otherwise there will be error.
+See [Data formats](../../design/adr/device-service/0011-DeviceService-Rest-API.md#data-formats) for supported data type in EdgeX.
+
+### Device Profiles
+See [V2 Device Profile Reference](profile/V2DeviceProfileRef.md) for details, SDK now allows both YAML and JSON format.
```

Comment:
The referenced document doesn't describe the differences to assist in migration. Please add description here how to migrate profiles.

Reply:
Still need to detail what someone needs to change from V1 to V2 to migrate a profile.